### PR TITLE
Feature/cc 0017 a user can interact with the sidebar

### PR DESCRIPTION
**Pull request by BowenLi61**
cc-0017 A user can interact with the sidebar: When click the button , the button is highlighted, when click the button, the button is lifted up and highlight and the content box appear. when click the close button. the content box disappear and the button highlight effect disappeared.
sidebarcontent

### CODE REVIEW

File: src/components/EditorSideBar/EditorSideBarContent.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      w="356px"
+      h="calc(100vh - 72px)"
+      top="72px"
+      left="96px"
+      position="fixed"
+      z-index="10"
+      color="#fff"
+      p={6}
+    >
+      {props.iconClick}
```
Comment (tatoMa) [inline src/components/EditorSideBar/EditorSideBarContent.tsx:22]: This iconClick on here may not be a proper name for this variable. It is the components this side menu would render on the screen. Try to use some meaningful variable name for it.

File: src/components/EditorSideBar/index.tsx [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+              icon={item.icon}
+              onHandleIconClick={() => handleIconClick(item.title)}
+              iconClick={iconClick}
+            />
+          ))}
+        </Flex>
+      </Box>
+      {isOpen && (
+        <EditorSideBarContent iconClick={iconClick} onHandleCloseClick={handleCloseClick} />
+      )}
```
Comment (tatoMa) [inline src/components/EditorSideBar/index.tsx:37]: You can use some of the chakra built-in transition components to make this popup window animate smooth.
image

**Review by tatoMa — COMMENTED** [review]
Maybe consider clicking this icon menu twice to open and close the popup window? It's just my opinion.

File: src/layouts/index.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const Layout: React.FC<{ children: ReactNode }> = ({ children }) => {
         {children}
         <Footer />
       </Box>
-      <PriceBar />
+      {/* <PriceBar /> */}
```
Comment (Fuchih) [inline src/layouts/index.tsx:21]: I was wondering why you need to comment <PriceBar /> out ?

**Review by Fuchih — COMMENTED** [review]
I was wondering why you neet to delete those code?

**Comment by Fuchih** [comment]
> I was wondering why you neet to delete those code?

File: src/styles/theme.ts [hunk 4]
```diff
@@ -53,8 +53,6 @@ const components = {
     }),
   },
   Button: { ...buttonStyles },
-  Drawer: { ...priceBarDrawer },
-  DrawerButton,
 };
```
Comment (Fuchih) [inline src/styles/theme.ts:56]: I was wondering why you need to delete those code?

File: src/layouts/index.tsx (hunk 3 above)
Comment (stkwn) [reply src/layouts/index.tsx:21]: The PriceBar is the old one which will be on top of the layer and will block this popup box.

File: src/components/EditorSideBar/index.tsx (hunk 2 above)
Comment (stkwn) [reply src/components/EditorSideBar/index.tsx:37]: Talked with BA and might design it later.

File: src/components/EditorSideBar/EditorSideBarContent.tsx (hunk 1 above)
Comment (stkwn) [reply src/components/EditorSideBar/EditorSideBarContent.tsx:22]: Got it. Update with "iconClickTitle".

File: src/styles/theme.ts (hunk 4 above)
Comment (stkwn) [reply src/styles/theme.ts:56]: Update later. Might be caused by dealing with conflict wrongly.

File: src/layouts/index.tsx (hunk 3 above)
Comment (Fuchih) [reply src/layouts/index.tsx:21]: It should be new one, please doubt check

File: src/components/EditorSideBar/EditorSideBarContent.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+const SideBarContainer = (props: Props) => {
+  return (
+    <Box
+      bg="background.secondary"
+      w="356px"
+      h="calc(100vh - 72px)"
+      top="72px"
+      left="96px"
+      position="fixed"
+      z-index="10"
```
Comment (Fuchih) [inline src/components/EditorSideBar/EditorSideBarContent.tsx:18]: It should be zIndex not z-index

File: src/components/EditorSideBar/EditorSideBarContent.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        as="button"
+        onClick={props.onHandleCloseClick}
+        position="absolute"
+        top="calc(48% - 84px)"
+        right="-26px"
+        h="168px"
+        w="56px"
+        bg="background.secondary"
+        clipPath="polygon(0% 0%, 100% 8%, 100% 92%, 0% 100%)"
+        z-index="1"
```
Comment (Fuchih) [inline src/components/EditorSideBar/EditorSideBarContent.tsx:33]: as above(zIndex)

Comment (stkwn) [reply src/components/EditorSideBar/EditorSideBarContent.tsx:33]: OK. Thank you.

File: src/styles/theme.ts (hunk 4 above)
Comment (stkwn) [reply src/styles/theme.ts]: already update and fixed it.

**Review by tatoMa — APPROVED** [review]
LGTM for now. New UI design may need to discuss for later card.

**Review by Fuchih — APPROVED** [review]

File: src/styles/theme.ts [hunk 7]
```diff
@@ -59,4 +60,4 @@ const components = {
 
 const theme = extendTheme({ colors, config, styles, components});
 
-export default theme;
+export default theme;
```
Comment (devharris7) [inline src/styles/theme.ts:63]: eof

File: src/components/EditorSideBar/index.tsx [hunk 8]
```diff
@@ -1,15 +1,45 @@
...
-    <Box bg="background.primary" w="98px" h="100vh" position="fixed" top="72px" left="0">
-      <Flex align="center" justify="center" flexDirection="column" maxW="98px">
-        {sideBarItemList.map((item) => (
-          <EditorSideBarItem key={item.title} title={item.title} icon={item.icon} />
-        ))}
-      </Flex>
+    <Box >
+      <Box bg="background.primary" w="98px" h="100vh" position="fixed" top="72px" left="0">
+        <Flex align="center" justify="center" flexDirection="column" maxW="98px">
+          {sideBarItemList.map((item) => (
```
Comment (devharris7) [inline src/components/EditorSideBar/index.tsx:29]: consider wrap this part separately with `React.useMemo`

Comment (devharris7) [reply src/components/EditorSideBar/index.tsx:29]: React.memo()

React.useCallback()

File: src/components/EditorSideBar/index.tsx [hunk 9]
```diff
@@ -1,15 +1,45 @@
...
+            />
+          ))}
+        </Flex>
+      </Box>
+      {isOpen && (
+        <EditorSideBarContent iconClickTitle={iconClickTitle} onHandleCloseClick={handleCloseClick} />
+      )}
     </Box>
   );
 };
```
Comment (devharris7) [inline src/components/EditorSideBar/index.tsx:45]: export default React.memo(EditorSideBar);

Comment (devharris7) [reply src/components/EditorSideBar/index.tsx:45]: react developer tool - profiler

**Pull request merged**